Mainnet/Identity/IdentityContracts/IdentityProvider.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Stratis.SmartContracts;
+
+/// <summary>
+/// A constrained version of ERC780.
+/// This contract only holds claims from one identity provider, the "Owner".
+/// </summary>
+public class IdentityProvider : SmartContract
+{
+    private Address Owner
+    {
+        get
+        {
+            return this.PersistentState.GetAddress(nameof(Owner));
+        }
+        set
+        {
+            this.PersistentState.SetAddress(nameof(Owner), value);
+        }
+    }
+
+    public IdentityProvider(ISmartContractState state) : base(state)
+    {
+        this.Owner = Message.Sender;
+    }
+
+    public void ChangeOwner(Address newOwner)
+    {
+        Assert(this.Owner == Message.Sender);
+        this.Owner = newOwner;
+    }
+
+    public void AddClaim(Address issuedTo, uint topic, byte[] data)
+    {
+        Assert(this.Owner == Message.Sender);
+
+        byte[] oldData = this.GetClaim(issuedTo, topic);
+        bool exists = oldData != null;
+
+        // This exact claim value is stored already. Nothing to change.
+        if (exists && ByteArrayCompare(oldData, data))
+            return;
+
+        this.SetClaim(issuedTo, topic, data);
+
+        if (exists)
+        {
+            this.Log(new ClaimChanged
+            {
+                IssuedTo = issuedTo,
+                Topic = topic,
+                Data = data
+            });
+        }
+        else
+        {
+            this.Log(new ClaimAdded
+            {
+                IssuedTo = issuedTo,
+                Topic = topic,
+                Data = data
+            });
+        }
+    }
+
+    public void RemoveClaim(Address issuedTo, uint topic)
+    {
+        Assert(this.Owner == Message.Sender);
+
+        // Nothing to delete.
+        byte[] oldData = this.GetClaim(issuedTo, topic);
+        if (oldData == null)
```

Comment:
Value can never be null here because `GetBytes` will return `byte[0]` for empty values. Should be fine to remove, or check `oldData.Length == 0`